src/deadline/unreal_submitter/unreal_open_job/unreal_open_job.py

```
@@ -0,0 +1,842 @@
+            name=None if shared_settings.name in ["", "Untitled"] else shared_settings.name,
+            steps=steps,
+            environments=[
+                UnrealOpenJobEnvironment.from_data_asset(env) for env in data_asset.environments
+            ],
+            extra_parameters=[
+                UnrealOpenJobParameterDefinition.from_unreal_param_definition(param)
+                for param in data_asset.get_job_parameters()
+            ],
+            job_shared_settings=JobSharedSettings.from_u_deadline_cloud_job_shared_settings(
+                shared_settings
+            ),
+        )
+
+    @staticmethod
+    def serialize_template(template: Template):
+        template_json = json.loads(template.json(exclude_none=True))
+        ordered_keys = [
+            "specificationVersion",
+            "name",
+            "parameterDefinitions",
+            "jobEnvironments",
+            "steps",
+        ]
+        ordered_data = dict(
+            OrderedDict((key, template_json[key]) for key in ordered_keys if key in template_json)
+        )
+        return ordered_data
+
+    @staticmethod
+    def update_job_parameter_values(
+        job_parameter_values: list[dict[str, Any]],
+        job_parameter_name: str,
+        job_parameter_value: Any,
+    ) -> list[dict[str, Any]]:
+        param = next((p for p in job_parameter_values if p["name"] == job_parameter_name), None)
+        if param:
+            param["value"] = job_parameter_value
+        return job_parameter_values
+
+    def _create_missing_extra_parameters_from_template(self):
+        try:
+            extra_param_names = [p.name for p in self._extra_parameters]
+            for p in self.get_template_object()["parameterDefinitions"]:
+                if p["name"] not in extra_param_names:
+                    self._extra_parameters.append(UnrealOpenJobParameterDefinition.from_dict(p))
+        except FileNotFoundError:
```

Why are we ignoring this error

---

Because this additional functionality only trying to fill parameters missed in data assets. It should not break the entire building process

---

Could we add a comment to that effect?